library/core/src/slice/sort.rs

```diff
@@ -457,9 +457,9 @@ where
             //  - `offsets_l` contains valid offsets into `v` collected during the partitioning of
             //    the last block, so the `l.offset` calls are valid.
             unsafe {
-                end_l = end_l.offset(-1);
-                ptr::swap(l.offset(*end_l as isize), r.offset(-1));
-                r = r.offset(-1);
+                end_l = end_l.sub(1);
+                ptr::swap(l.add(*end_l as usize), r.sub(1));
```

Comment:
Not this PR, but consider following along to use `usize::from(*end_l)` here -- I went off to check that `*end_l` wasn't signed, and if it was a `from` then it'd be obvious that it's unsigned (since the `from` wouldn't compile for signed -> unsigned).